tagstudio/src/core/library.py

```diff
@@ -499,10 +501,14 @@ def open_library(self, path: str | Path) -> int:
                     self.verify_ts_folders()
                     major, minor, patch = json_dump["ts-version"].split(".")
 
-                    # Load Extension Blacklist ---------------------------------
-                    if "ignored_extensions" in json_dump.keys():
-                        self.ignored_extensions = json_dump["ignored_extensions"]
-
+                    # Load Extension Exclude List ---------------------------------
+                    # The following if statment is only used when a old library is loaded
+                    if "ignored_extensions" in json_dump:
+                        self.ext_list = json_dump["ignored_extensions"]
```

Comment:
This line is why mypy complains but it keeps the ignored_extensions and saves them as the ext_list to update old librarys.
Should i remove this line and break it or does someone know a way arround it?

Reply:
keep `ignored_extensions` in `JsonLibary` (sic!) for backward compatibility and change the loading here to:
```
self.ext_list = json_dump.get("ext_list") or json_dump.get("ignored_extensions")
```
so it will first try to use `ext_list` and in case it doesnt exist yet (old library), it will try to use `ignored_extensions`